scripts/ImportExportTest.sh

```diff
@@ -0,0 +1,302 @@
+            rm expected.json obtained.json
+            rm -f stderr.txt
+        elif [ "${IMPORT_TEST_TYPE}" == "evm-assembly" ]
+        then
+            local types=( "asm" "bin" "bin-runtime" "opcodes" "srcmap" "srcmap-runtime" )
+            local _TESTED=1
+            if ! $SOLC --combined-json bin,bin-runtime,opcodes,asm,srcmap,srcmap-runtime --pretty-json "$nth_input_file" "${all_input_files[@]}" > expected.json 2> expected.error
+            then
+                printf "\n"
+                echo "$nth_input_file"
+                cat expected.error
+                UNCOMPILABLE=$((UNCOMPILABLE + 1))
+                return 0
+            else
+                for contract in $(jq '.contracts | keys | .[]' expected.json 2> /dev/null)
+                do
+                    for type in "${types[@]}"
+                    do
+                        jq --raw-output ".contracts.${contract}.\"${type}\"" expected.json > "expected.${type}"
+                    done
+
+                    assembly=$(cat expected.asm)
+                    if [ "$assembly" != "" ] && [ "$assembly" != "null" ]
+                    then
+                        if ! $SOLC --combined-json bin,bin-runtime,opcodes,asm,srcmap,srcmap-runtime --pretty-json --import-asm-json expected.asm > obtained.json 2> obtained.error
+                        then
+                            printf "\n"
+                            echo "$nth_input_file"
+                            cat obtained.error
+                            FAILED=$((FAILED + 1))
+                            return 0
+                        else
+                            for type in "${types[@]}"
+                            do
+                                for obtained_contract in $(jq '.contracts | keys | .[]' obtained.json  2> /dev/null)
+                                do
+                                    jq --raw-output ".contracts.${obtained_contract}.\"${type}\"" obtained.json > "obtained.${type}"
+                                    set +e
+                                    DIFF="$(diff "expected.${type}" "obtained.${type}")"
+                                    set -e
+                                    if [ "$DIFF" != "" ]
+                                    then
+                                        if [ "$DIFFVIEW" == "" ]
+                                        then
+                                            echo -e "ERROR: JSONS differ for $1: \n $DIFF \n"
+                                            echo "Expected:"
+                                            cat  "expected.${type}"
+                                            echo "Obtained:"
+                                            cat "obtained.${type}"
+                                        else
+                                            # Use user supplied diff view binary
+                                            $DIFFVIEW expected.json obtained.json
+                                        fi
```

Comment:
We have a [`diff_values()`](https://github.com/ethereum/solidity/blob/v0.8.13/scripts/common.sh#L174-L184) helper. How about adding `diff_files()` next to it and moving this part there so that it's reusable? You could then also add support for `$DIFFVIEW` to `diff_values()`.

Reply:
Ah true, I still need to add `$DIFFVIEW` support for `diff_values`.